annotator-core/src/main/java/edu/ucr/cs/riple/core/cache/Impact.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 Nima Karimipour
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.core.cache;
+
+import com.google.common.collect.ImmutableSet;
+import edu.ucr.cs.riple.core.metadata.index.Error;
+import edu.ucr.cs.riple.core.metadata.index.Fix;
+import edu.ucr.cs.riple.injector.location.Location;
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** Stores the set of errors which will be triggered if the containing fix is applied. */
+public class Impact {
+
+  /** Target fix. */
+  public final Fix fix;
+  /** Set of triggered errors, if this fix is applied to source code. */
+  protected ImmutableSet<Error> triggeredErrors;
+  /** Set of triggered fixes on downstream dependencies if containing fix is applied. */
+  protected ImmutableSet<Fix> triggeredFixesOnDownstream;
```

Comment:
Why do we say triggered fixes _on_ downstream dependencies?  I thought we cannot apply fixes to downstream dependencies, only the target?

Reply:
The comment in `Report.java` is clearer:

https://github.com/ucr-riple/NullAwayAnnotator/blob/0f1ca3271126ace7500e83cdeb6c9ef4d28515da/annotator-core/src/main/java/edu/ucr/cs/riple/core/Report.java#L55-L59

Can we rename these fields to `triggeredFixesFromDownstreamErrors` for clarity?

Reply:
No, we are not applying any fix to downstream dependencies, renamed for clarity, thank you for the suggestion. c4100c0fe640fe656945968b720720f98b799b99

Reply:
Sorry, happened in another branch ce46c8ffb53747f41330ad098b550f37fa21f90b